apps/next-product-starter/src/lib/add-cue-note.ts

```diff
@@ -18,21 +18,26 @@ const addCueNote =
 
     await axios
       .post(`/api/lessons/notes/${resource.slug}`, {
-        text: cue.text,
+        text: cueNote.text,
         startTime: context.currentTime,
-        endTime: context.currentTime + readingTime(cue.text).time / 1000,
+        endTime: context.currentTime + readingTime(cueNote.text).time / 1000,
         state: context.writingCueNoteVisibility,
         contact_id: context.viewer.contact_id,
         image: context.viewer.avatar_url,
       })
       .catch(() => {
         send('FAIL')
       })
-      .then(() => {
-        send('DONE_SUBMITTING_NOTE')
+      .then(({data}: any) => {
+        // pass cue text with props to get an id right
+        // away from response and avoid making another request
+        const cueNoteText = `{"id":"${data.id}","text":"${data.text}","type":"${data.type}","image":"${data.image}","startTime":${data.start_time},"endTime":${data.end_time}}`
```

Comment:
suggestion: this is hard for me to parse visually and understand what is actually occurring. It looks like the primary transformation is with the start/end time, which might mean we should consider making it consistent everywhere but I'd consider it like this as a more readable approach:

```suggestion
        const {start_time: startTime, end_time:endTime, ...noteData} = data
        const cueNoteText = JSON.stringify({
          startTime,
          endTime,
          ...noteData
        })
```

with this approach I am clearly identify the transformation and using standard API to create the string data we need to build the `VTTCue`